### ISSUE TITLE
Typing space doesn't work in Gantt task editing when Keyboard navigation enabled

### ISSUE THREAD

**Issue by matsakyan** [telerik/kendo-ui-core#4563]
### Bug report

In Gantt Chart during editing cell after clicking in edited input pressing 'space' key doesn't work.

NOTE: the issue appears only when 'Keyboard navigation' activated

### Reproduction of the problem
1. open this kendo [example](https://demos.telerik.com/kendo-ui/gantt/keyboard-navigation)
1. open title cell editor by double-clicking in some title (or by pressing 'enter')
1. click in some position of title to change the cursor position
1. try to type space by pressing 'space' key

### Current behavior
typing space doesn't work

NOTE: Clicking on the cell once again focuses the input and it is possible to type space. Then clicking again on the cell focuses the cell and it is not possible to type space. Thus, every odd click would not allow space and every even click allows it.
The issue comes from Gantt keyboard navigation and it is reproducible with R3 2020 as well.

### Expected/desired behavior
must be able to type space

### Environment
* **Browser:** all

**Comment by Dimitar-Goshev** [comment]
Not reproducible on demos with 2023.3.1114
